src/catkin_pkg/package.py

```diff
@@ -177,8 +177,17 @@ def validate(self, warnings=None):
         # accepting upper case letters and hyphens only for backward compatibility
         if not re.match('^[a-zA-Z0-9][a-zA-Z0-9_-]*$', self.name):
             errors.append('Package name "%s" does not follow naming conventions' % self.name)
-        elif not re.match('^[a-z][a-z0-9_]*$', self.name):
-            new_warnings.append('Package name "%s" does not follow the naming conventions. It should start with a lower case letter and only contain lower case letters, digits and underscores.' % self.name)
+        else:
+            if self.has_buildtool_depend_on_catkin():
```

Comment:
I don't have a strong feeling one way or the other, but `build_type` seems purpose built for this, but I thought maybe I wasn't considering some corner case which asks for this style check instead.